polkadot/xcm/xcm-executor/src/config.rs

```diff
@@ -114,4 +115,14 @@ pub trait Config {
 
 	/// Transactional processor for XCM instructions.
 	type TransactionalProcessor: ProcessTransaction;
+
+	/// Allows optional logic execution for the `HrmpNewChannelOpenRequest` XCM notification sent by
+	/// the relay chain.
+	type HrmpNewChannelOpenRequestHandler: HandleHrmpNewChannelOpenRequest;
+	/// Allows optional logic execution for the `HrmpChannelAccepted` XCM notification sent by the
+	/// relay chain.
+	type HrmpChannelAcceptedHandler: HandleHrmpChannelAccepted;
+	/// Allows optional logic execution for the `HrmpChannelClosing` XCM notification sent by the
+	/// relay chain.
+	type HrmpChannelClosingHandler: HandleHrmpChannelClosing;
```

Comment:
```suggestion
	/// Allows optional logic execution for the `HrmpNewChannelOpenRequest` XCM notification.
	type HrmpNewChannelOpenRequestHandler: HandleHrmpNewChannelOpenRequest;
	/// Allows optional logic execution for the `HrmpChannelAccepted` XCM notification.
	type HrmpChannelAcceptedHandler: HandleHrmpChannelAccepted;
	/// Allows optional logic execution for the `HrmpChannelClosing` XCM notification.
	type HrmpChannelClosingHandler: HandleHrmpChannelClosing;
```

As @franciscoaguirre told me, the concept of "sent by the relay chain" should be handled by a barrier, something similar to this: [link](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/barriers.rs#L352-L354) .

I would say`AllowHrmpChannelNotificationsFromRelayChain` or `AllowHrmpChannelNotificationsFrom` with origin filter - @franciscoaguirre which one do you prefer more?

Reply:
Should that barrier setting be introduced in this PR?

Reply:
no worries, I do it as a part of my iissue: https://github.com/paritytech/polkadot-sdk/issues/3692 when setting handler for system parachain runtimes